### ISSUE TITLE
The Collapsible Listbox example includes neither aria-controls nor aria-owns

### ISSUE THREAD

**Issue by craigkovatch** [w3c/aria-practices#1125]
Shouldn't the triggering button for a menu popup or a dropdown list reference the overlaid element with `aria-controls` or `aria-owns`? I admit I am unclear about which would be appropriate -- the overlaid element is not a child of the `<button>`, but neither does the button particularly control the popup. Perhaps this is why its excluded, but it seems strange that there is no accessible relationship between the button and the popup at all. Could someone help me understand better?

**Comment by mcking65** [comment]
@@craigkovatch wrote:
> Shouldn't the triggering button for a menu popup or a dropdown list reference the overlaid element with `aria-controls` or `aria-owns`? 

If it is a button that opens a menu, the [menu button pattern](http://w3c.github.io/aria-practices/#menubutton) includes the following text in the "WAI-ARIA Roles, States, and Properties" section:

>Optionally, the element with role button  has a value specified for aria-controls that refers to the element with role menu . 

@@craigkovatch wrote:
>I admit I am unclear about which would be appropriate -- the overlaid element is not a child of the `<button>`, 

Correct, aria-owns would not be appropriate, especially since a button has children presentational true and thus cannot have interactive, focusable descendants.

>but neither does the button particularly control the popup. 

Actually it does, so aria-controls is appropriate.

>it seems strange that there is no accessible relationship between the button and the popup at all.

Agree. I don't recall why we do not have aria-controls in the collapsable listbox example. It may be an oversight. That would be my mistake on this particular example.

That said, the collapsable listbox example has been a somewhat illustrative or experimental example as we grapple with how to best code the ARIA equivalent of a HTML select with size=1. This is a topic that will get more attention before our next publication. It is related to several open ARIA issues, including:

* [w3c/html-aam#46](https://github.com/w3c/html-aam/issues/46): select element should map to listbox role regardless of presentation
* w3c/aria#721: Add aria-valuetext as a supported property of listbox
* w3c/aria#722: Add aria-expanded as a supported property of listbox
* w3c/aria#817: Update combobox spec as per html spec/html-aam mapping for `<select>`    * [HTML-AAM: select element should map to listbox role regardless of presentation
* w3c/aria#998: The Combobox 1.0 design pattern references incorrectly include aria-owns instead of aria-controls.

While this example is functional and meets a specific need, it has short comings. For instance, it does not support aria-required. Nonetheless, I think it is likely we will significantly revise it. How we do so depends  on how the above issues get resolved.

In the meantime, if you implement a widget that models it, it would certainly be appropriate to use aria-controls on the button. Note, btw, that there is also significant discussion of the value of aria-controls itself (w3c/aria#995).

If this does not answer your question, please feel free to re-open the issue.

**Comment by craigkovatch** [comment]
@mcking65 thank you for the detailed thoughts. My remaining question is this:
>>but neither does the button particularly control the popup.

>Actually it does, so aria-controls is appropriate.

My understanding of `aria-controls` is for e.g. a `role=combobox` where focus remains on an element (in this case an `<input>`) but keyboard events *control* some other element, in this case a `role=listbox` and probably with `aria-activedescendant`.

Or, it is used for e.g. multimedia controls, like a `button` that controls a video playback element elsewhere in the document. Again, focus remains on the button, but the results of pushing it affect another "distant" element.

In the case of a button with a listbox popup, focus presumably moves onto/into the listbox itself, rather than remaining on the button. So `aria-controls` here is confusing to me. 

I too have been grappling for a while with the right way to represent the equivalent of an HTML `<select>` of size 1.

**Comment by craigkovatch** [comment]
@mcking65 I don't believe I have permissions to re-open the issue so tagging you for visibility

**Comment by mcking65** [comment]
@craigkovatch, the net with aria-controls is that the specification is ambiguous and we need to clear it up.

Whether or not focus moves hasn't, thus far, been one of the criteria for determining whether or not to apply aria-controls. Frankly, we do not have clear criteria.

For now, the best I can say is mimic use in the APG patterns. This is an issue we intend to address in ARIA 1.2.

**Comment by craigkovatch** [comment]
@mcking65 OK thank you. Is this a discussion I can join? Having implemented several patterns for similar-but-different controls now, would love to at least be a fly on the wall, and hopefully contribute to the discussion.

**Comment by mcking65** [comment]
Oops, I was wrong ... we have punted aria-controls to aria 1.3. However, we may have to partially tackle it to resolve combobox issues. At any rate, the center of gravity for discussion is w3c/aria#995.
